### ISSUE TITLE
Copy/paste messes up correctly formatted code

### ISSUE THREAD

**Issue by abarth** [flutter/flutter-intellij#677]
## Steps to Reproduce

1. Open a dart file in a Flutter project.
2. Copy the following to the clipboard:

```dart
  double estimateScrollOffsetExtent({
    int firstIndex,
    int lastIndex,
    double leadingScrollOffset,
    double trailingScrollOffset,
  });
```

3. Paste into the body of a class declaration.
4. Notice that you get the following text:

```dart
  double estimateScrollOffsetExtent({
  int firstIndex,
  int lastIndex,
  double leadingScrollOffset,
  double trailingScrollOffset,
  });
```

The pasted text has removed two spaces of indent each from lines 2 through 5.

## Version info

```
[✓] IntelliJ IDEA Community Edition (version 2016.3)
    • Dart plugin version 163.9166.22
    • Flutter plugin version 0.1.8.1
```

**Comment by pq** [comment]
FYI @stevemessick

**Comment by stevemessick** [comment]
Cannot reproduce.

@alexander-doroshko Is there a preference that might affect this? Or could it be fixed in the current dev sources? (2017.1 EAP)

**Comment by devoncarew** [comment]
I do see this in 2016.3.3 (dart plugin 163.11306).

**Comment by alexander-doroshko** [comment]
This is an abstract method with optional parameters, right? Yes, looks like Dart plugin has a bug with multiline optional parameters formatting. Please file it n our tracker.

Workaround 1: use Paste Simple (Cmd+Alt+Shift+V) instead of standard Paste.

Workaround 2: change Reformat On Paste option in Preferences | Editor | General | Smart Keys (default is Indent Each Line).

**Comment by abarth** [comment]
> Please file it n our tracker.

Where is your your tracker?  /me is a noob.

**Comment by alexander-doroshko** [comment]
https://youtrack.jetbrains.com/issues/WEB, Create Issue button, set Subsystem to Dart.

**Comment by abarth** [comment]
Done: https://youtrack.jetbrains.com/issue/WEB-25187

Thanks!

**Comment by alexander-doroshko** [comment]
I've published the Dart plugin with the fix for IntelliJ IDEA 2016.3.x. You should get a suggestion to update the plugin or try Help | Check for updates.

**Comment by stevemessick** [comment]
Thanks for the quick update @alexander-doroshko !

I confirmed that paste works correctly for this example with all four "Reformat on paste" settings.
